clients/fides-js/src/lib/tcf.ts

```diff
@@ -158,6 +158,29 @@ export const generateFidesString = async ({
         }
       });
 
+      // Set legitimate interest for special-purpose only vendors
+      if (!experience.minimal_tcf && experience.gvl?.vendors) {
```

Comment:
I don't think we should/need to check for minimal experience here. It should work for both minimal and full experiences. If not, we may need to tweak the minimal to get this to work since we want to support opting in/out without the full payload being ready yet.